quickwit/quickwit-common/src/tower/buffer.rs

```diff
@@ -166,10 +166,7 @@ impl<R> fmt::Debug for BufferLayer<R> {
 
 impl<R> Clone for BufferLayer<R> {
```

Comment:
`#[derive(Copy, Clone)]` did not work here?

Reply:
No, the derived `Clone` implementation unnecessarily adds `Clone` bounds on `S` and `R`:
```rust
#[automatically_derived]
        impl<S: ::core::clone::Clone, R: ::core::clone::Clone> ::core::clone::Clone
        for Buffer<S, R>
        where
            S: Service<R>,
        {
            #[inline]
            fn clone(&self) -> Buffer<S, R> {
                Buffer {
                    bound: ::core::clone::Clone::clone(&self.bound),
                    inner: ::core::clone::Clone::clone(&self.inner),
                }
            }
        }
```